UploadObject.java

```diff
@@ -0,0 +1,42 @@
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.SdkClientException;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+
+import java.io.File;
+import java.io.IOException;
+
+public class UploadObject {
+
+    public static void main(String[] args) throws IOException {
+        Regions clientRegion = Regions.DEFAULT_REGION;
+        String bucketName = "*** Bucket name ***";
+        String stringObjKeyName = "*** String object key name ***";
+        String fileObjKeyName = "*** File object key name ***";
+        String fileName = "*** Path to file to upload ***";
+
+        
+            //This code expects that you have AWS credentials set up per:
+            // https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/setup-credentials.html
+            BasicAWSCredentials creds = new BasicAWSCredentials("AKIASLH77DNM7TFJOIPT", "5cTiK5meMxNN4U8qmvtaRQw98XYirwPx0OiHit56"); 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This looks like an AWS access key.
Don&#39;t embed access keys directly into code. The AWS SDKs and the AWS Command Line Tools allow you to put access keys in known locations so that you do not have to keep them in code. Learn more about [Best Practices for Managing AWS Access Keys](https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html).